test/utils/k8s/service.go

```diff
@@ -0,0 +1,86 @@
+package k8s
+
+import (
+	"fmt"
+	"github.com/skupperproject/skupper/test/utils/constants"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"time"
+)
+
+func WaitForServiceToBeAvailableDefaultTimeout(ns string, kubeClient kubernetes.Interface, name string) (service *apiv1.Service, err error) {
+	return WaitForServiceToBeAvailable(ns, kubeClient, name, constants.SkupperServiceReadyPeriod)
+}
+
+func WaitForServiceToBeAvailable(ns string, kubeClient kubernetes.Interface, name string, timeout time.Duration) (service *apiv1.Service, err error) {
+
+	// Create a service informer
+	done := make(chan struct{})
+	factory := informers.NewSharedInformerFactory(kubeClient, 0)
+	svcInformer := factory.Core().V1().Services().Informer()
+	svcInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			svc, _ := cache.MetaNamespaceKeyFunc(obj)
+			// if service has been found, close the done channel
+			if svc == fmt.Sprintf("%s/%s", ns, name) {
+				service = obj.(*apiv1.Service)
+				close(done)
+			}
+		},
+	})
+	stop := make(chan struct{})
+	go svcInformer.Run(stop)
+
+	// Wait for informer to be done or a timeoutCh
+	timeoutCh := time.After(timeout)
+	select {
+	case <-timeoutCh:
+		err = fmt.Errorf("timed out waiting on service to be available: %s", name)
+	case <-done:
+		break
+	}
+
+	// stop informer
+	close(stop)
+
+	return
```

Comment:
And it is returning nothing when it should return (service *apiv1.Service, err error)...
But this is compiling! is github showing the wrong diff?

Reply:
Same as above.